File: python/filter1.py
import numpy as np


def filter1(g, u):
    '''
    Normal filtering
    :param g: impulse
    :param u: signal
    :return:
    '''
    ii = len(u)
    l = len(g)
    m = int(np.floor(l / 2))
    start = np.array([u[0] for _ in range(m)])
    end = np.array([u[-1] for _ in range(m)])
    y = np.concatenate((start, u, end))
    y = np.convolve(g, y)
    return y[l - 1:l + ii - 1]

File: matlab/filter1.m
function x=filter1(g,u);

%Normal filtering
%u - signal
%g - impulse response
%x=filter1(g,u);

ii=length(u);
L=length(g); 
m=floor(L/2);
y=[u(ones(1,m)) u u(ii*ones(1,m))];
y=conv(g,y);
x=y(L:L+ii-1);
   